osu.Server.Queues.ScorePump/Performance/PerformanceProcessor.cs

```diff
@@ -0,0 +1,374 @@
+                    {
+                        key.BeatmapId,
+                        key.RulesetId,
+                        key.ModValue
+                    }, transaction: transaction)).ToArray();
+            }
+
+            if (rawDifficultyAttributes == null)
+                return null;
+
+            DifficultyAttributes difficultyAttributes = LegacyRulesetHelper.CreateDifficultyAttributes(ruleset.RulesetInfo.OnlineID);
+            difficultyAttributes.FromDatabaseAttributes(rawDifficultyAttributes.ToDictionary(a => (int)a.attrib_id, a => (double)a.value), new APIBeatmap
+            {
+                ApproachRate = beatmap.diff_approach,
+                DrainRate = beatmap.diff_drain,
+                OverallDifficulty = beatmap.diff_overall,
+                CircleCount = beatmap.countNormal,
+                SliderCount = beatmap.countSlider,
+                SpinnerCount = beatmap.countSpinner
+            });
+
+            return difficultyAttributes;
+        }
+
+        /// <summary>
+        /// Retrieves a beatmap from the database.
+        /// </summary>
+        /// <param name="beatmapId">The beatmap's ID.</param>
+        /// <param name="connection">The <see cref="MySqlConnection"/>.</param>
+        /// <param name="transaction">An existing transaction.</param>
+        /// <returns>The retrieved beatmap, or <c>null</c> if not existing.</returns>
+        public async Task<Beatmap?> GetBeatmapAsync(int beatmapId, MySqlConnection connection, MySqlTransaction? transaction = null)
+        {
+            if (beatmapCache.TryGetValue(beatmapId, out var beatmap))
+                return beatmap;
+
+            return beatmapCache[beatmapId] = await connection.QuerySingleOrDefaultAsync<Beatmap?>($"SELECT * FROM {Beatmap.TABLE_NAME} WHERE `beatmap_id` = @BeatmapId", new
+            {
+                BeatmapId = beatmapId
+            }, transaction: transaction);
+        }
+
+        /// <summary>
+        /// Updates a user's stats with their total PP/accuracy.
+        /// </summary>
+        /// <remarks>
+        /// This does not insert the new stats values into the database.
+        /// </remarks>
+        /// <param name="rulesetId">The ruleset for which to update the total PP.</param>
+        /// <param name="userStats">An existing <see cref="UserStats"/> object to update with.</param>
+        /// <param name="connection">The <see cref="MySqlConnection"/>.</param>
+        /// <param name="transaction">An existing transaction.</param>
+        public async Task UpdateUserStatsAsync(UserStats userStats, int rulesetId, MySqlConnection connection, MySqlTransaction? transaction = null)
+        {
+            List<SoloScoreWithPerformance> scores = (await connection.QueryAsync<SoloScoreWithPerformance>(
+                $"SELECT `s`.*, `p`.`pp` AS `pp` FROM {SoloScore.TABLE_NAME} `s` "
```

Comment:
```suggestion
                $"SELECT `s`.*, `p`.`pp` FROM {SoloScore.TABLE_NAME} `s` "
```

Pretty sure the alias is unnecessary